### PR TITLE
Add overlayed v0.3.0 (new cask)

### PR DESCRIPTION
**Pull request by Hacksore**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/o/overlayed.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  url "https://github.com/Hacksore/overlayed/releases/download/v#{version}/overlayed_#{version}_universal.dmg",
+      verified: "github.com/Hacksore/overlayed/"
+  name "Overlayed"
+  desc "Modern, open-source, and free voice chat overlay for Discord"
+  homepage "https://overlayed.dev/"
+
+  livecheck do
+    url :url
+    strategy :github_latest
+  end
```
Comment (p-linnane) [inline Casks/o/overlayed.rb:14]: Unless your repo is using GitHub pre-releases, or has unstable releases that use regular `X.Y.Z` tags, we can remove this. Livecheck will use a `:git` strategy for GitHub releases by default without needing to specify it.

**Review by p-linnane — COMMENTED** [review]
Thanks for your PR! Just one question/suggestion.

File: Casks/o/overlayed.rb (hunk 1 above)
Comment (Hacksore) [reply Casks/o/overlayed.rb]: I don't know much about how the livecheck works so if omitting it will do the same thing I'm fine with that.

**Review by p-linnane — APPROVED** [review]
Looks good. Thanks for contributing to Homebrew!

**Pull request merged**